### PR TITLE
Hide unsupported modes

### PR DESCRIPTION
**Pull request by cquiroz**
We can now load the modes directly from the odb rather than from a file

### CODE REVIEW

File: explore/src/main/scala/explore/cache/ModesCache.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import clue.StreamingClient
+import explore.DefaultErrorPolicy
+import explore.modes.SpectroscopyModesMatrix
+import fs2.Stream
+import japgolly.scalajs.react.*
+import lucuma.react.common.ReactFnProps
+import lucuma.schemas.ObservationDB
+import queries.common.ModesQueriesGQL
+
+case class ModesCache(
```
Comment (cquiroz) [inline explore/src/main/scala/explore/cache/ModesCache.scala:18]: I wonder if this is overkill for this case

Comment (toddburnside) [reply explore/src/main/scala/explore/cache/ModesCache.scala:18]: Do you mean caching the modes might be overkill?

Comment (cquiroz) [reply explore/src/main/scala/explore/cache/ModesCache.scala:18]: We want to call it only once not  every time we open the modees tile, in that sense caching is important
However they also don't (often) change upstream, there is no subscription for example

Comment (toddburnside) [reply explore/src/main/scala/explore/cache/ModesCache.scala:18]: We'll have to figure this out when there is filtering on the ODB for CfP, resource availability, etc. But for now this is 👍

File: model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala [hunk 2]
```diff
@@ -332,65 +337,61 @@ object SpectroscopyModeRow {
...
-      }
 
-  given CellDecoder[Option[SpectroscopyCapabilities]] =
-    CellDecoder.stringDecoder
-      .map {
-        case "Nod&Shuffle" => SpectroscopyCapabilities.NodAndShuffle.some
-        case "coronagraph" => SpectroscopyCapabilities.Coronagraphy.some
-        case _             => none
-      }
+  // decodores for instruments are used locally as they are not lawful
```
Comment (toddburnside) [inline model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala:341]: Spelling of `decoders`.

**Review by toddburnside — APPROVED** [review]
LGTM. Thanks for doing this!

**Pull request merged**